### PR TITLE
New: IPTable extensions

### PR DESCRIPTION
**Pull request by sibicramesh**
--> Added a new field `Extensions` in iprule structure which is of type `[]string`, where the rules are explicitly set
-- [project](https://github.com/orgs/aporeto-inc/projects/395)

### CODE REVIEW

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "extensions",
    "component": "enforcerd",
    "pr-id": "1122",
    "commit-sha": "804a4e0738f020ec765a75d1ebee26a9c2b911b6"
  },
  {
    "project": "extensions",
    "component": "trireme-lib",
    "pr-id": "775",
    "commit-sha": "737b40232858cdbbe6c842f0b1ca3cac952e6423"
  }
]
```

File: controller/internal/supervisor/iptablesctrl/acls.go [hunk 1]
```diff
@@ -287,6 +293,27 @@ func (i *Instance) generateACLRules(contextID string, rule *aclIPset, chain stri
 	return iptRules, reverseRules
 }
 
+// programExtensionsRules programs iptable rules for the given instruction
+func (i *Instance) programExtensionsRules(rule *aclIPset, chain, proto string) error {
+
+	if byteCodes, ok := rule.extensions["bpf"]; ok {
+		for _, byteCode := range byteCodes {
```
Comment (dstiliadis) [inline controller/internal/supervisor/iptablesctrl/acls.go]: I think we can make this more generic. Don't need to just be the bpf extension. It is essentially a list of parameters that can be blidly passed to iptables without us looking at them. Opaque at this point.Essentially the caller will provide a string array that will include "-m" "bpf" ... or whatever else .. we don't need to hardwire these values.

File: controller/internal/supervisor/iptablesctrl/iptables.go [hunk 2]
```diff
@@ -561,10 +561,11 @@ type ipsetInfo struct {
...
 type aclIPset struct {
-	ipset     string
-	ports     []string
-	protocols []string
-	policy    *policy.FlowPolicy
+	ipset      string
+	ports      []string
+	protocols  []string
+	extensions map[string][]string
+	policy     *policy.FlowPolicy
```
Comment (dstiliadis) [inline controller/internal/supervisor/iptablesctrl/iptables.go:568]: see above. I would just make this a []string .. and pass them to iptables .. we could keep it as a map with the idea that we will type and customize some of them in the future. So, there are "raw" entries that we just pass along and there can be typed entries that we know how to handle.

Comment (sibicramesh) [reply controller/internal/supervisor/iptablesctrl/iptables.go:568]: Why do you want to have two fields when we can do this in one ?. We can have the map key as `raw` and pass all its values blindly to the iptables, so the code looks something like this with both functionalities

```
		if byteCodes, ok := rule.extensions["bpf"]; ok {
		for _, byteCode := range byteCodes {
			rulesspec := []string{
				"-p", proto,
				"--match", "multiport", "--dports", strings.Join(rule.ports, ","),
				"-m", "bpf",
				"--bytecode", byteCode,
				"-j", "DROP",
			}
			if err := i.ipt.Append(i.appPacketIPTableContext, chain, rulesspec...); err != nil {
				return fmt.Errorf("unable to program rule for extension bpf: %v", err)
			}
		}
	}

	if rules, ok := rule.extensions["raw"]; ok {
		if err := i.ipt.Append(i.appPacketIPTableContext, chain, rules...); err != nil {
			return fmt.Errorf("unable to program rule for extension bpf: %v", err)
		}
	}
	
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "extensions",
    "component": "enforcerd",
    "pr-id": "1122",
    "commit-sha": "f69d2855b04480df27789f0c3ed5e6da37428faf"
  },
  {
    "project": "extensions",
    "component": "trireme-lib",
    "pr-id": "775",
    "commit-sha": "d1f9d51b0912a21bdf5918c4738b599b980fdf2f"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "extensions",
    "component": "enforcerd",
    "pr-id": "1122",
    "commit-sha": "f69d2855b04480df27789f0c3ed5e6da37428faf"
  },
  {
    "project": "extensions",
    "component": "trireme-lib",
    "pr-id": "775",
    "commit-sha": "51f861bb39f3971e9b950450299290160fe52c29"
  }
]
```

**Review by dstiliadis — DISMISSED** [review]

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "extensions",
    "component": "trireme-lib",
    "pr-id": "775",
    "commit-sha": "589cd6903730a8d57a37fa39d8c954421fd69ec2"
  },
  {
    "project": "extensions",
    "component": "enforcerd",
    "pr-id": "1122",
    "commit-sha": "64274261f70f983a41664a90928d5b5b981b3a65"
  }
]
```

**Pull request merged**